### DISCUSSION TITLE
Enforcing Date/Time Constraint Help!

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#614] [Help]
I have a scenario where my client is providing events to attend. These events only span one day - i.e. start and finish on the same day. The DB (currently - but can be changed) holds values for the date of the event, the time it starts and the time it ends (i.e. "date", "from", "to"). The migration for these fields are as follows:

```
$table->date('date');
$table->time('from');
$table->time('to');
```

Since these events start and end on the same day, I want a mechanism whereby the user keying in event details cannot erroneously provide an event that spans multiple days. DateTimePicker doesn't provide an option to just provide a UI for entering a "time". Currently, there are only two options available to me:

1. Provide "date" with DatePicker and then use TextInputs to require the user to enter in a time with the format "HH:MM". This works and stores correctly but when the field is rendered back we get something like "2021-09-27T09:00:00.000000Z" and since I am using a TextInput I can't format this correctly.

2. Change the DB so that it stores two dates signifiying the start and end of the event however this will not prevent a user entering an event that spans multiple days.

Any help appreciated! Project still in dev so can amend DB structure if needed. Open to ideas!

**Comment by danharrin** [comment]
Hey, are you using the admin panel or the form builder?

If you're using the admin panel: I'd suggest submitting a PR for a new `TimePicker`, which extends `DateTimePicker` and disables the date part.

If you're using the form builder: You could submit a PR for `TimePicker`, or instead you could use [input masking](https://filamentadmin.com/docs/2.x/forms/fields#input-masking) to enforce the `HH:MM` format. Input masking features will be available to the admin panel when we release v2 later this year.

**Comment by titantwentyone** [comment]
Thanks Dan. It's the admin panel. I'll take a look at a PR.

**Reply by titantwentyone** [reply]
New PR #618. Any feedback welcome. Bit green around the gills re: PRs! :)
